src/database/world_graph.lua

```diff
@@ -511,7 +517,7 @@ function worldGraph:collect(preCollectedItems)
         table.insert(_i, i)
       end
     end
-    for i, v in ipairs(_i) do table.remove(availableLocations, i) end
+    for i = #_i, 1, -1 do table.remove(availableLocations, _i[i]) end
```

Comment:
?

Reply:
What I gather is that the code is trying to remove all of the locations from `availableLocations` whose items are about to be collected (in the next few lines below this). However, it's currently removing the wrong values from the table -- it should be removing `v`, not `i`, and even then, it still doesn't account for all of the higher-indexed values shifting over by one after each removal (which is why I remove them in descending order).

Reply:
There are actually much more efficient ways of doing this, but I initially chose to just implement the most straightforward "direct" fix.